Shared/Samples/Authenticate with OAuth/AuthenticateWithOAuthView.swift

```diff
@@ -0,0 +1,96 @@
+// Copyright 2023 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import SwiftUI
+import ArcGIS
+import ArcGISToolkit
+
+struct AuthenticateWithOAuthView: View {
+    /// The authenticator to handle authentication challenges.
+    @StateObject private var authenticator = Authenticator(
+        oAuthUserConfigurations: [.arcgisDotCom]
+    )
+
+    init() {
+        // Sets authenticator as ArcGIS and Network challenge handlers to handle authentication
+        // challenges.
+        ArcGISEnvironment.authenticationManager.handleChallenges(using: authenticator)
+    }
+    
+    /// The map to be displayed on the map view.
+    @State private var map: Map = {
+        // The portal to authenticate with named user.
+        let portal = Portal(url: .portal, connection: .authenticated)
+        
+        // The portal item to be displayed on the map.
+        let portalItem = PortalItem(
+            portal: portal,
+            id: .trafficMap
+        )
+        
+        // Creates map with portal item.
+        return Map(item: portalItem)
+    }()
+    
+    var body: some View {
+        MapView(map: map)
+            .authenticator(authenticator)
+            .onDisappear {
+                // Reset challenge handlers.
+                ArcGISEnvironment.authenticationManager.handleChallenges(using: nil)
+
+                // Sign out.
```

Comment:
```suggestion
```
I think this method name is self explanatory.

Reply:
fixed